pybossa/api/task.py

```diff
@@ -102,6 +102,8 @@ def _preprocess_post_data(self, data):
         if 'n_answers' not in data:
             project = project_repo.get(project_id)
             data['n_answers'] = project.get_default_n_answers()
+            if not project:
+                raise BadRequest(f'Non existing project id {project_id}')
```

Comment:
Are any other details of the request available inside `data[]` that would help identify the caller? If so, those could be included in the exception details.

Reply:
Please see the ticket comments for planned work on the gigwork listener to add identification for the caller. Thanks!